### DISCUSSION TITLE
Sync Shared Libraries without having to authenticate again.

### DISCUSSION THREAD

**Discussion by titusjan** [abraunegg/onedrive#1951] [Ideas]
I am trying to use to your marvelous program to make a mirror/backup of all the SharePoint Shared Libraries in our organization. I have made a small PowerShell script that returns a list of names of all the Shared Libraries I want to sync, so I've got that part automated.

Now I run into the issue that I cannot make a script to sync a new Shared Library because, according to [the documenation](https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#4-authenticate-the-client), I need to create a new config file per Shared Library and authenticate again. Creating a new config file is not a problem, but the authentication step can't be automated because you need to enter your password in a web browser.

I'd like to be able to sync a new Shared Library without having to authenticate myself again for each shared library. A one time authentication is fine of course. Is it possible that you implement this?

Is the authentication step really necessary? Can you please explain why or why not?

**Comment by abraunegg** [comment]
@titusjan 
> I'd like to be able to sync a new Shared Library without having to authenticate myself again for each shared library.

You can simply copy the 'refresh_token' from your already authenticated configuration to your new configuration as the token is 100% independent of the application configuration to access the SharePoint sites. This 'token' tells the API you are authenticated - the embedded authentication scopes govern however what you can or cannot access.

However - if you do not keep this token fresh (which the application does when the token is used), it will get stale, thus, require re-authentication.

> Is the authentication step really necessary? Can you please explain why or why not?

Yes it is necessary - it is your key to unlock the lock provided by Microsoft. They created the lock and the process to provide the key to unlock.

**Reply by titusjan** [reply]
Thank you for your quick reply. I will try copying the refresh token.

**Reply by titusjan** [reply]
Copying the refresh token works. Thanks.
